botocore/handlers.py

```diff
@@ -328,9 +328,8 @@ def _sse_md5(params, sse_member_prefix='SSECustomer'):
     key_as_bytes = params[sse_key_member]
     if isinstance(key_as_bytes, str):
         key_as_bytes = key_as_bytes.encode('utf-8')
-    key_md5_str = base64.b64encode(get_md5(key_as_bytes).digest()).decode(
-        'utf-8'
-    )
+    md5_val = get_md5(key_as_bytes, usedforsecurity=False).digest()
```

Comment:
This md5 parameter [maps to](https://github.com/boto/botocore/blob/8cb37cd9d449ac1a3ea2c70fbb5ff3f94d9d7b45/botocore/data/s3/2006-03-01/service-2.json#L2366-L2371) the `x-amz-server-side​-encryption​-customer-key-MD5 ` header, which according to [the S3 docs](https://docs.aws.amazon.com/AmazonS3/latest/userguide/ServerSideEncryptionCustomerKeys.html) is used for integrity checks, not security:
> Use this header to provide the base64-encoded 128-bit MD5 digest of the encryption key according to [RFC 1321](http://tools.ietf.org/html/rfc1321). Amazon S3 uses this header for a message integrity check to ensure that the encryption key was transmitted without error.